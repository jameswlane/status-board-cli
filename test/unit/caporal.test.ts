describe('src/index.ts', () => {
  it('should match our snapshot', () => {
    expect(true).toBeTruthy();
  });
});
