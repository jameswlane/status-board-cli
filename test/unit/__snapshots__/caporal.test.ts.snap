// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/index.ts should match our snapshot 1`] = `
Program {
  "_autocomplete": Autocomplete {
    "_completions": Object {},
    "_program": [Circular],
    "_tab": Complete {
      "_completions": Array [],
      "_events": Object {},
      "_eventsCount": 0,
      "_maxListeners": undefined,
      "_results": Array [],
      "domain": null,
      "options": Object {
        "cache": false,
        "name": "jest",
        "ttl": 300000,
      },
    },
  },
  "_bin": "jest",
  "_commands": Array [
    Command {
      "_action": [Function],
      "_alias": null,
      "_args": Array [
        Argument {
          "_default": undefined,
          "_description": "type of component",
          "_name": "component",
          "_synopsis": "<component>",
          "_type": "required",
          "_validator": Validator {
            "_program": [Circular],
            "_validator": Array [
              "widget",
              "dashboard",
              "job",
            ],
          },
          "_variadic": false,
          "default": [Function],
          "description": [Function],
          "synopsis": [Function],
        },
        Argument {
          "_default": undefined,
          "_description": "name of component",
          "_name": "name",
          "_synopsis": "<name>",
          "_type": "required",
          "_validator": null,
          "_variadic": false,
          "default": [Function],
          "description": [Function],
          "synopsis": [Function],
        },
      ],
      "_default": false,
      "_description": "generates a basic component of type widget, dashboard or job",
      "_lastAddedArgOrOpt": Argument {
        "_default": undefined,
        "_description": "name of component",
        "_name": "name",
        "_synopsis": "<name>",
        "_type": "required",
        "_validator": null,
        "_variadic": false,
        "default": [Function],
        "description": [Function],
        "synopsis": [Function],
      },
      "_logger": Object {
        "_events": Object {},
        "_eventsCount": 0,
        "_maxListeners": undefined,
        "_names": Array [
          "caporal",
        ],
        "debug": [Function],
        "domain": null,
        "emitErrs": false,
        "error": [Function],
        "exceptionHandlers": Object {},
        "exitOnError": true,
        "filters": Array [],
        "id": null,
        "info": [Function],
        "level": "info",
        "levels": Object {
          "debug": 4,
          "error": 0,
          "info": 2,
          "silly": 5,
          "verbose": 3,
          "warn": 1,
        },
        "padLevels": false,
        "profilers": Object {},
        "rewriters": Array [],
        "silly": [Function],
        "stripColors": false,
        "transports": Object {
          "caporal": CaporalTransport {
            "_onError": [Function],
            "level": "info",
            "name": "caporal",
          },
        },
        "verbose": [Function],
        "warn": [Function],
      },
      "_name": "generate",
      "_options": Array [],
      "_program": [Circular],
      "debug": [Function],
      "description": [Function],
      "error": [Function],
      "info": [Function],
      "log": [Function],
      "warn": [Function],
    },
    Command {
      "_action": [Function],
      "_alias": null,
      "_args": Array [
        Argument {
          "_default": undefined,
          "_description": "App to deploy",
          "_name": "app",
          "_synopsis": "<app>",
          "_type": "required",
          "_validator": Validator {
            "_program": [Circular],
            "_validator": /\\^myapp\\|their-app\\$/,
          },
          "_variadic": false,
          "default": [Function],
          "description": [Function],
          "synopsis": [Function],
        },
        Argument {
          "_default": "local",
          "_description": "Environment to deploy on",
          "_name": "env",
          "_synopsis": "[env]",
          "_type": "optional",
          "_validator": Validator {
            "_program": [Circular],
            "_validator": /\\^dev\\|staging\\|production\\$/,
          },
          "_variadic": false,
          "default": [Function],
          "description": [Function],
          "synopsis": [Function],
        },
      ],
      "_default": false,
      "_description": "generates a new dashboard project",
      "_lastAddedArgOrOpt": Argument {
        "_default": "local",
        "_description": "Environment to deploy on",
        "_name": "env",
        "_synopsis": "[env]",
        "_type": "optional",
        "_validator": Validator {
          "_program": [Circular],
          "_validator": /\\^dev\\|staging\\|production\\$/,
        },
        "_variadic": false,
        "default": [Function],
        "description": [Function],
        "synopsis": [Function],
      },
      "_logger": Object {
        "_events": Object {},
        "_eventsCount": 0,
        "_maxListeners": undefined,
        "_names": Array [
          "caporal",
        ],
        "debug": [Function],
        "domain": null,
        "emitErrs": false,
        "error": [Function],
        "exceptionHandlers": Object {},
        "exitOnError": true,
        "filters": Array [],
        "id": null,
        "info": [Function],
        "level": "info",
        "levels": Object {
          "debug": 4,
          "error": 0,
          "info": 2,
          "silly": 5,
          "verbose": 3,
          "warn": 1,
        },
        "padLevels": false,
        "profilers": Object {},
        "rewriters": Array [],
        "silly": [Function],
        "stripColors": false,
        "transports": Object {
          "caporal": CaporalTransport {
            "_onError": [Function],
            "level": "info",
            "name": "caporal",
          },
        },
        "verbose": [Function],
        "warn": [Function],
      },
      "_name": "new",
      "_options": Array [],
      "_program": [Circular],
      "debug": [Function],
      "description": [Function],
      "error": [Function],
      "info": [Function],
      "log": [Function],
      "warn": [Function],
    },
    Command {
      "_action": [Function],
      "_alias": null,
      "_args": Array [],
      "_default": false,
      "_description": "lists all available components (widgets or jobs) among all packages",
      "_lastAddedArgOrOpt": null,
      "_logger": Object {
        "_events": Object {},
        "_eventsCount": 0,
        "_maxListeners": undefined,
        "_names": Array [
          "caporal",
        ],
        "debug": [Function],
        "domain": null,
        "emitErrs": false,
        "error": [Function],
        "exceptionHandlers": Object {},
        "exitOnError": true,
        "filters": Array [],
        "id": null,
        "info": [Function],
        "level": "info",
        "levels": Object {
          "debug": 4,
          "error": 0,
          "info": 2,
          "silly": 5,
          "verbose": 3,
          "warn": 1,
        },
        "padLevels": false,
        "profilers": Object {},
        "rewriters": Array [],
        "silly": [Function],
        "stripColors": false,
        "transports": Object {
          "caporal": CaporalTransport {
            "_onError": [Function],
            "level": "info",
            "name": "caporal",
          },
        },
        "verbose": [Function],
        "warn": [Function],
      },
      "_name": "list",
      "_options": Array [],
      "_program": [Circular],
      "debug": [Function],
      "description": [Function],
      "error": [Function],
      "info": [Function],
      "log": [Function],
      "warn": [Function],
    },
    Command {
      "_action": [Function],
      "_alias": null,
      "_args": Array [
        Argument {
          "_default": undefined,
          "_description": "runs atlasboard in port",
          "_name": "port",
          "_synopsis": "[port]",
          "_type": "optional",
          "_validator": null,
          "_variadic": false,
          "default": [Function],
          "description": [Function],
          "synopsis": [Function],
        },
      ],
      "_default": false,
      "_description": "starts Status Board's server",
      "_lastAddedArgOrOpt": Option {
        "_booleanFlag": undefined,
        "_default": undefined,
        "_description": "loads only dashboards matching",
        "_long": "--dashboard",
        "_longCleanName": "dashboard",
        "_name": "dashboard",
        "_program": [Circular],
        "_required": false,
        "_short": undefined,
        "_shortCleanName": undefined,
        "_synopsis": "--dashboard <dashboard>",
        "_validator": null,
        "_valueType": "required",
        "_varadic": false,
        "default": [Function],
        "description": [Function],
        "synopsis": [Function],
      },
      "_logger": Object {
        "_events": Object {},
        "_eventsCount": 0,
        "_maxListeners": undefined,
        "_names": Array [
          "caporal",
        ],
        "debug": [Function],
        "domain": null,
        "emitErrs": false,
        "error": [Function],
        "exceptionHandlers": Object {},
        "exitOnError": true,
        "filters": Array [],
        "id": null,
        "info": [Function],
        "level": "info",
        "levels": Object {
          "debug": 4,
          "error": 0,
          "info": 2,
          "silly": 5,
          "verbose": 3,
          "warn": 1,
        },
        "padLevels": false,
        "profilers": Object {},
        "rewriters": Array [],
        "silly": [Function],
        "stripColors": false,
        "transports": Object {
          "caporal": CaporalTransport {
            "_onError": [Function],
            "level": "info",
            "name": "caporal",
          },
        },
        "verbose": [Function],
        "warn": [Function],
      },
      "_name": "start",
      "_options": Array [
        Option {
          "_booleanFlag": undefined,
          "_default": undefined,
          "_description": "skips npm package install (faster startup)",
          "_long": "--noinstall",
          "_longCleanName": "noinstall",
          "_name": "noinstall",
          "_program": [Circular],
          "_required": false,
          "_short": undefined,
          "_shortCleanName": undefined,
          "_synopsis": "--noinstall",
          "_validator": null,
          "_valueType": undefined,
          "_varadic": undefined,
          "default": [Function],
          "description": [Function],
          "synopsis": [Function],
        },
        Option {
          "_booleanFlag": undefined,
          "_default": undefined,
          "_description": "runs only jobs matching 'myjob'",
          "_long": "--job",
          "_longCleanName": "job",
          "_name": "job",
          "_program": [Circular],
          "_required": false,
          "_short": undefined,
          "_shortCleanName": undefined,
          "_synopsis": "--job <job>",
          "_validator": null,
          "_valueType": "required",
          "_varadic": false,
          "default": [Function],
          "description": [Function],
          "synopsis": [Function],
        },
        Option {
          "_booleanFlag": undefined,
          "_default": undefined,
          "_description": "loads only dashboards matching",
          "_long": "--dashboard",
          "_longCleanName": "dashboard",
          "_name": "dashboard",
          "_program": [Circular],
          "_required": false,
          "_short": undefined,
          "_shortCleanName": undefined,
          "_synopsis": "--dashboard <dashboard>",
          "_validator": null,
          "_valueType": "required",
          "_varadic": false,
          "default": [Function],
          "description": [Function],
          "synopsis": [Function],
        },
      ],
      "_program": [Circular],
      "debug": [Function],
      "description": [Function],
      "error": [Function],
      "info": [Function],
      "log": [Function],
      "warn": [Function],
    },
    Command {
      "_action": [Function],
      "_alias": null,
      "_args": Array [],
      "_default": false,
      "_description": "install package dependencies",
      "_lastAddedArgOrOpt": null,
      "_logger": Object {
        "_events": Object {},
        "_eventsCount": 0,
        "_maxListeners": undefined,
        "_names": Array [
          "caporal",
        ],
        "debug": [Function],
        "domain": null,
        "emitErrs": false,
        "error": [Function],
        "exceptionHandlers": Object {},
        "exitOnError": true,
        "filters": Array [],
        "id": null,
        "info": [Function],
        "level": "info",
        "levels": Object {
          "debug": 4,
          "error": 0,
          "info": 2,
          "silly": 5,
          "verbose": 3,
          "warn": 1,
        },
        "padLevels": false,
        "profilers": Object {},
        "rewriters": Array [],
        "silly": [Function],
        "stripColors": false,
        "transports": Object {
          "caporal": CaporalTransport {
            "_onError": [Function],
            "level": "info",
            "name": "caporal",
          },
        },
        "verbose": [Function],
        "warn": [Function],
      },
      "_name": "install",
      "_options": Array [],
      "_program": [Circular],
      "debug": [Function],
      "description": [Function],
      "error": [Function],
      "info": [Function],
      "log": [Function],
      "warn": [Function],
    },
  ],
  "_defaultCommand": null,
  "_description": "Mocked description",
  "_helper": Help {
    "_customHelp": Object {
      "generate": "atlasboard generate widget mywidget",
      "install": "status-board install",
      "list": "status-board list",
      "new": "atlasboard new mywallboard",
      "start": "my help for the order command",
    },
    "_program": [Circular],
  },
  "_logger": Object {
    "_events": Object {},
    "_eventsCount": 0,
    "_maxListeners": undefined,
    "_names": Array [
      "caporal",
    ],
    "debug": [Function],
    "domain": null,
    "emitErrs": false,
    "error": [Function],
    "exceptionHandlers": Object {},
    "exitOnError": true,
    "filters": Array [],
    "id": null,
    "info": [Function],
    "level": "info",
    "levels": Object {
      "debug": 4,
      "error": 0,
      "info": 2,
      "silly": 5,
      "verbose": 3,
      "warn": 1,
    },
    "padLevels": false,
    "profilers": Object {},
    "rewriters": Array [],
    "silly": [Function],
    "stripColors": false,
    "transports": Object {
      "caporal": CaporalTransport {
        "_onError": [Function],
        "level": "info",
        "name": "caporal",
      },
    },
    "verbose": [Function],
    "warn": [Function],
  },
  "_supportedShell": Array [
    "bash",
    "zsh",
    "fish",
  ],
  "_version": "1.0.0",
  "bin": [Function],
  "description": [Function],
  "logger": [Function],
  "name": [Function],
  "version": [Function],
}
`;
